server/server.go

```diff
@@ -3356,3 +3356,12 @@ func (s *Server) setFirstPingTimer(c *client) {
 	d += time.Duration(addDelay)
 	c.ping.tmr = time.AfterFunc(d, c.processPingTimer)
 }
+
+func (s *Server) updateRemoteSubscription(c *client, sub *subscription) {
```

Comment:
Sorry for mentioning that now, but since we are just removing here, the name of the function should reflect that, or, maybe better, keep this name but add a `delta int` as the last parameter of this function (and pass `delta` in updateRouteSubscriptionMap(), gatewayUpdateSubInterest() and updateLeafNodes().

In the places you called updateRemoteSubscription() in this PR you would pass `-1`. There are only 1 or 2 places in the rest of the server code where we could possibly call this function with a positive delta, because in some cases we do some of the things based on the client's kind (CLIENT, etc..).
We internally need to have a deeper look at that, but I think having this function with a `delta` could make it be reusable later when we cleanup the rest of the code.

Reply:
Good input! I will fix asap

Reply:
@kozlovic  fixed.